apigw/src/enduser/maproutes.ts

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import express, { Router, Request } from 'express'
+import axios from 'axios'
+
+const router = Router()
+
+router.get('/map-api/autocomplete', (req: Request, res) => {
```

Comment:
Suggestion: use a proxy based on `express-http-proxy` instead of custom endpoints. You can add the extra header easily using `proxyReqOptDecorator`

These custom solutions throw away all response headers and the post endpoint does unnecessary extra work just to parse the proxied body into memory